snapshot/git/gitdb/db_test.go

```diff
@@ -0,0 +1,167 @@
+package gitdb
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/snapshot"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+func TestIngestDir(t *testing.T) {
+	fixture, err := setup()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ingestDir, err := fixture.tmp.TempDir("ingest_dir")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	contents := []byte("bar")
+	filename := filepath.Join(ingestDir.Dir, "foo.txt")
+
+	if err = ioutil.WriteFile(filename, contents, 0777); err != nil {
+		t.Fatal(err)
+	}
+
+	id, err := fixture.db.IngestDir(ingestDir.Dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	path, err := fixture.db.Checkout(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	actual, err := ioutil.ReadFile(filename)
```

Comment:
Is this checking against the right location? I would've expected join(path, foo.txt). Also I take it that it's a TODO to use the fixture's two commits?